### ISSUE TITLE
Layers are deleted from the map but are not deleted from layerGroupProvider

### ISSUE THREAD

**Issue by AnatolyStrakhov** [yagajs/leaflet-ng2#365]
Hi All,
I use image as a map and put other layers (marker, circle marker, geojson) on that. HTML skeleton of component  looks like that
 ```
  <yaga-map>
    <yaga-zoom-control>
      <yaga-layers-control>
        <yaga-layer-group yaga-overlay-layer="Tags">
          <ul>
            <li *ngFor="let point of points">
              <yaga-circle-marker>
                <yaga-tooltip></yaga-tooltip>
              </yaga-circle-marker>
            </li>
          </ul>
        </yaga-layer-group>
        <yaga-image-overlay>
        </yaga-image-overlay>
      </yaga-layers-control>
    </yaga-zoom-control>
  </yaga-map>
```

The points is an `@Input()` of my component. When I change image and points, yaga-map renders it correctly. But if disable and enable the layer, points from all images are shown.
**Scenario**.
1. Open _image1_ with _points1_. Map is correct, _points1_ are on the _image1_.
2. Switch to _image2_ and _points2_ by changing inputs of the component. Map is correct, _points2_ are on the _image2_.
3. Disable the group layer of points (using the drop-down layer control on the map). Map is correct, _image2_ has no points.
4. Enable the group layer (using the drop-down layer control on the map). Map is incorrect, _image2_ has both _points1_ and _points 2_.
The root cause is layers are deleted from the map but are not deleted from `layerGroupProvider`. 
_marker.directive.js_ for example
In the constructor it has

`layerGroupProvider.ref.addLayer(_this);`

I changed it to

```
layerGroupProvider.ref.addLayer(_this);
_this.layerGroupProvider = layerGroupProvider;

```
Then I see in `ngOnDestroy` that `layerGroupProvider` keeps layers already deleted from the map.
I added `removeLayer()` here.

```
  MarkerDirective.prototype.ngOnDestroy = function() {
    this.removeFrom(this._map);
    this.layerGroupProvider.ref.removeLayer(this);
  };

```
And it solves the issue. Could you fix it please?
Please be sure that other directives are affected too.

**Comment by atd-schubert** [comment]
Oh thanks! It seems we forgot to adjust the destroy part during the introduction of Angular providers. I will try to fix it as soon as possible...

**Comment by AnatolyStrakhov** [comment]
Fixed in 1.0.0-rc10. Thank you.
